### PR TITLE
Plans 2023: Update discounted price styling

### PR DESCRIPTION
**Pull request by aneeshd16**
Related to Automattic/martech#1472

## Proposed Changes

* This PR updates the styling of discounts according to xkhhUlDowF13dRoXJqlRDM-fi-1872-54424 . It builds on top of #74960 so that it can utilize the refactored `PlanFeatures2023GridHeaderPrice` component.

## Testing Instructions

* Confirm that the discount styling matches the design in the following cases:
  * On `/start/plans` for a currency with a first-year discount.
image
image

  * On `/plans/<site slug>` where pro-rated discounts are applicable.
 
image

  * Both the above cases after switching to the IDR currency.
  
image
 
image

* Also check for mobile/tablet resolutions:
image
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/plan-comparison-grid.tsx [hunk 1]
```diff
@@ -323,7 +325,6 @@ const PlanComparisonGridHeaderCell: React.FunctionComponent<
 		'is-current-plan': current,
 	} );
 	const rawPrice = planPropertiesObj.rawPrice;
-	const isLargeCurrency = rawPrice ? rawPrice > 99000 : false;
```
Comment (aneeshd16) [inline client/my-sites/plan-features-2023-grid/plan-comparison-grid.tsx:334]: This implementation is incorrect as it calculates the value of `isLargeCurrency` for each plan separately. So, it is possible that the value is `false` for the Personal plan, but for the Business plan, the value is `true`. This causes the `is-large-currency` class to apply to only some of the plans displayed, leading to inconsistent styling.

File: client/my-sites/plan-features-2023-grid/style.scss [hunk 2]
```diff
@@ -323,6 +323,10 @@
 			font-weight: 600;
 		}
 
+		&.is-bottom-aligned {
+			vertical-align: bottom;
+		}
+
```
Comment (aneeshd16) [inline client/my-sites/plan-features-2023-grid/style.scss:329]: The `is-bottom-aligned` class was already applied, but this CSS rule was missing.

File: client/my-sites/plan-features-2023-grid/billing-timeframe.tsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ function usePerMonthDescription( {
 	if ( ! isMonthlyPlan ) {
 		const fullTermPriceText =
 			currencyCode && null !== maybeDiscountedFullTermPrice
-				? formatCurrency( maybeDiscountedFullTermPrice, currencyCode )
+				? formatCurrency( maybeDiscountedFullTermPrice, currencyCode, { stripZeros: true } )
```
Comment (aneeshd16) [inline client/my-sites/plan-features-2023-grid/billing-timeframe.tsx:53]: Small change.
Before:
image

After:
image

File: client/my-sites/plan-features-2023-grid/style.scss [hunk 4]
```diff
@@ -336,27 +340,25 @@
 
 	.plan-features-2023-grid__pricing {
```
Comment (chriskmnds) [inline client/my-sites/plan-features-2023-grid/style.scss]: WDYT about consolidating all the `plan-features-2023-grid__pricing` styles into one location while at it. So either move everything from here into `header-price` (which most look like overrides of `PlanPrice`) or move the [new styles](https://github.com/Automattic/wp-calypso/pull/74961/files#diff-0b8c173504dad73d4da3beb1c69d587a74c61893c99961bc0544b5bd506eb1b7R15-R22) here?

**Review by chriskmnds — APPROVED** [review]
Thanks @aneeshd16. Tests out. I left a suggestion for a refactor, which I'm also happy to do it, and push the commit.

The only somewhat minor inconsistency I notice is the display and cross-outs when having decimals in the price e.g.

Screenshot 2023-03-28 at 1 24 16 PM

Screenshot 2023-03-28 at 1 39 14 PM


- the alignment is not identical between the two
- the double strikethrough feels a little odd (which looks like something that was previously solved with the diagonal line?)

File: client/my-sites/plan-features-2023-grid/style.scss (hunk 4 above)
Comment (aneeshd16) [reply client/my-sites/plan-features-2023-grid/style.scss]: Agreed!

**Comment by aneeshd16** [comment]
> The only somewhat minor inconsistency I notice is the display and cross-outs when having decimals in the price e.g.
> * the alignment is not identical between the two
> * the double strikethrough feels a little odd (which looks like something that was previously solved with the diagonal line?)

Good catch! I'll figure out a way to replicate decimal pricing here.

**Comment by chriskmnds** [comment]
> Good catch!

By accident :D

**Comment by aneeshd16** [comment]
Hi @chriskmnds, I have consolidated all styles with the component via https://github.com/Automattic/wp-calypso/pull/74961/commits/8c335502fee873a0930d0175c9473322e3075af5. This does not address the issue with the decimal prices, I have asked for clarification in p1680077449183959-slack-C040UAN6TPS

**Comment by aneeshd16** [comment]
Since the `isLargeCurrency` prop must be computed at a higher level, it would require more refactoring. I'll tackle that separately in a follow-up PR. This PR is good to go from my end. 👍

**Comment by chriskmnds** [comment]
Sounds good, @aneeshd16. Thanks for addressing the feedback 👍 

p.s. it needs a rebase. I guess we'll ship the base PR first and then this, or can it go separately into `trunk`? Obviously, as you prefer. This also served well for testing the underlying refactors. I plan on shipping https://github.com/Automattic/wp-calypso/pull/74960 a little later or tomorrow - replied to your comments there in case you have other suggestions :-)

**Comment by aneeshd16** [comment]
Thanks @chriskmnds! Let's ship the base PR first, and I'll rebase this one after that. I'll have a look at https://github.com/Automattic/wp-calypso/pull/74960 soon!

**Comment by chriskmnds** [comment]
LGTM after rebase

**Pull request merged**